src/include/migraphx/op/pooling.hpp

```diff
@@ -64,12 +64,13 @@ struct pooling
         for(size_t i = 0; i < kdims; i++)
         {
             assert(lengths[i] <= input_lens[i + 2] + 2 * padding[i]);
+            std::size_t len = (ceil_mode)
+                                  ? ceil_divide<std::ptrdiff_t>(
+                                        input_lens[i + 2] + 2 * padding[i] - lengths[i], stride[i])
```

Comment:
Can you assign this to a local variable and then pass it to the function? This will avoid repeating this argument twice.

Reply:
fixed